openshift_day2_configurator/configurators/ingress.py

```diff
@@ -0,0 +1,377 @@
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.hazmat.primitives import serialization, hashes
+from cryptography import x509
+from cryptography.x509.oid import NameOID
+from datetime import datetime, timedelta
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+import os
+import base64
+import logging
+from typing import Any, Dict, Union, Optional
+from kubernetes.dynamic import DynamicClient
+
+from ocp_resources.resource import ResourceEditor
+from ocp_resources.configmap import ConfigMap
+from ocp_resources.secret import Secret
+from ocp_resources.pod import Pod
+from ocp_resources.ingress_controller import IngressController
+from ocp_resources.proxy import Proxy
+
+from openshift_day2_configurator.utils.general import (
+    verify_and_execute_configurator,
+)
+from rich.progress import Progress, TaskID
+from openshift_day2_configurator.utils.resources import create_ocp_resource
+
+CREATE_NEW_INGRESS_CERTIFICATE: str = "Create new Ingress certificate"
+CREATE_INGRESS_CERTIFICATE_CONFIGMAP: str = "Create ConfigMap with the new Ingress certificate"
+UPDATE_CLUSTER_PROXY_TRUSTED_CA: str = "Update cluster Proxy with the new Ingress certificate"
+CREATE_WILDCARD_CERTIFICATE_TLS_SECRET: str = "Create wildcard certificate tls secret"
+UPDATE_INGRESS_CONTROLLER_CERTIFICATE: str = "Update the IngressController certificate"
+WAIT_ON_INGRESS_PODS_RESCHEDULE: str = "Wait on Ingress pods to reschedule"
+
+OPENSHIFT_INGRESS_NAMESPACE: str = "openshift-ingress"
+INGRESS_CERTIFICATE_CONFIGMAP_NAME: str = "custom-ca"
+
+TIMEOUT_30MIN: int = 30 * 60
+
+
+def get_ingress_certificate_from_file(cluster_domain: str) -> str:
```

Comment:
no need to re-open the file x times, use caching

```
version = sys.version_info
if version[0] == 3 and version[1] < 9:
    from functools import lru_cache as cache
else:
    from functools import cache  # type: ignore[no-redef]
```

Reply:
done